alpha/apps/kaltura/lib/myPlaylistUtils.class.php

```diff
@@ -1091,19 +1091,41 @@ private static function updateEntryFilter(entryFilter $entry_filter, $partner_id
 	 */
 	protected static function getCaptionFilePath($captionAsset, &$localFilePath)
 	{
+		$local = false;
 		$captionFileSyncKey = $captionAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
-		list($captionFileSync, $local) = kFileSyncUtils::getReadyFileSyncForKey($captionFileSyncKey, true, false);
-		if (!$captionFileSync)
+		list ($captionFileSync, $path) = kFileSyncUtils::getFileSyncAndPathForFlavor($captionFileSyncKey, $captionAsset);
+
+		if(serveFlavorAction::$preferredStorageId)
```

Comment:
What if preferred id is set to 0 , will the actual results be as expected, better check if not null.